src/gui/properties/peerlistsortmodel.cpp

```diff
@@ -52,3 +68,8 @@ bool PeerListSortModel::lessThan(const QModelIndex &left, const QModelIndex &rig
         return QSortFilterProxyModel::lessThan(left, right);
     };
 }
+
+bool PeerListSortModel::filterAcceptsRow(const int sourceRow, const QModelIndex &sourceParent) const
+{
+    return !m_filteredRows.contains(sourceRow);
```

Comment:
I strongly dislike such a way.
IMO, the correct one is to get peer status from underlying model and check whether peer is connecting. It will also prevent you from invalidating filter for each peer when refreshing the data.

Reply:
this is a proxy model. this class doesn't know much about the data, all such info resides with peerlistwidget where the filtering is handled. your way may require whole redo

Reply:
>this is a proxy model. 

This isn't abstract proxy model. It still operates some data specific to concrete underlying model.
>your way may require whole redo

Well, maybe a little more work. If it scares you, then you shouldn't have taken this job.

Reply:
If this "feature" is generally approved, I will be able to implement "my way" myself, but it is unlikely to take any top place on my to-do list.

Reply:
> If it scares you, then you shouldn't have taken this job.

nowhere it said I've to deal with bickering russian

Reply:
@glassez wasn't the reply was to your comment that was totally unwarranted and had nothing to do with code or its supposed "flaws". You should carefully read what you write.

Anyway updated the implementation with the "glassez" way. All hail Mother Russia!

Reply:
> @glassez wasn't the reply was to your comment that was totally unwarranted and had nothing to do with code or its supposed "flaws".

Maybe if you take it out of context...

If we consider only the "tip of the iceberg", which I pointed out as one of the shortcomings of your original solution, and imagine that this is already existing code written once by someone, then this is quite similar to a potential target for one of your "optimize after measurement" PRs, since invalidating the filter for each peer within a loop clearly doesn't seem like the optimal way. Even if we go <del>"jagannatharjun" way (but I would still prefer not to go personal)</del> different way and implement an abstract filtering model that does not depend on a specific subject area, but simply filters out the specified rows, doesn't it look more optimal to set the entire set of rows in one call?

Reply:
> Maybe if you take it out of context...

Do tell me which universe's interpretation you're taking here.
I would like you to please take your own advice and refrain from passing unnecessary comments.